meta_client_v2/src/meta_impl.rs

```diff
@@ -0,0 +1,513 @@
+            };
+
+            if let Some(v) = receiver {
+                match self.fetch_action_cmd(v).await {
+                    Ok(()) => {
+                        info!(
+                            "Fetch cluster action cmd finished, cluster:{}",
+                            self.get_cluster_name()
+                        );
+                    }
+                    Err(err) => {
+                        error!(
+                            "Failed to get action cmd, cluster:{}, err:{}",
+                            self.get_cluster_name(),
+                            err
+                        );
+                        self.reconnect_heartbeat_channel().await;
+                    }
+                }
+            } else {
+                warn!("Skip action command fetch because no receiver found");
+            }
+
+            if time::timeout(self.error_wait_lease(), stop_rx.recv())
+                .await
+                .is_ok()
+            {
+                warn!("Receive exit message, exit the fetch_action_cmd loop");
+                break;
+            }
+        }
+    }
+
+    async fn fetch_action_cmd(
+        &self,
+        mut receiver: ClientDuplexReceiver<PbNodeHeartbeatResponse>,
+    ) -> Result<()> {
+        info!("Start fetch action cmd loop");
+        while let Some(resp) = receiver.try_next().await.context(FetchActionCmd)? {
+            info!(
+                "Fetch action cmd from meta, cluster:{}, action_cmd:{:?}",
+                self.get_cluster_name(),
+                resp,
+            );
+
+            let resp = NodeHeartbeatResponse::from(resp);
+            if let Err(e) = check_response_header(&resp.header) {
+                error!("Fetch action cmd failed, err:{}", e);
+                continue;
+            }
+            let event = match resp.action_cmd {
+                Some(action_cmd) => action_cmd,
+                None => {
+                    warn!("Fetch action cmd is empty, resp:{:?}", resp);
+                    continue;
+                }
+            };
+
+            if let Err(e) = self.handle_event(&event).await {
+                error!("Handler fail to handle event:{:?}, err:{}", event, e);
+            }
+        }
+
+        Ok(())
+    }
+
+    pub async fn handle_event(&self, event: &ActionCmd) -> Result<()> {
+        let handlers = self.handlers.read().await;
+        for handler in handlers.iter() {
+            if let Some(handler) = handler.upgrade() {
+                handler.handle(event).await.context(FailHandleEvent {
+                    name: handler.name().to_string(),
+                    event: event.clone(),
+                })?;
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Default meta client impl, will interact with a remote meta node.
+pub struct MetaClientImpl {
+    inner: Arc<Inner>,
+    runtime: Arc<Runtime>,
+
+    eventloop_handle: Mutex<Option<JoinHandle<()>>>,
+    stop_eventloop_tx: Mutex<Option<Sender<()>>>,
```

Comment:
stop_event_background_tx

Reply:
Not a good name and wait for a better name.